src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Packed.cs

```diff
@@ -357,12 +353,10 @@ private static int IndexOf<TNegator>(ref short searchSpace, short value, int len
                         Vector512<short> source0 = Vector512.LoadUnsafe(ref firstVector);
                         Vector512<short> source1 = Vector512.LoadUnsafe(ref oneVectorAwayFromEnd);
                         Vector512<byte> packedSource = PackSources(source0, source1);
-                        Vector512<byte> result = Vector512.Equals(packedValue, packedSource);
-                        result = NegateIfNeeded<TNegator>(result);
 
-                        if (result != Vector512<byte>.Zero)
+                        if (NegateIfNeeded<TNegator>(Vector512.EqualsAny(packedValue, packedSource)))
```

Comment:
Are the new checks cheaper than what we did earlier? Number of operations increased from 3 (Equals, NegateIfNeeded, !=) to 4 (NegateIfNeeded, EqualsAny, NegateIfNeeded, Equals).

Reply:
Its 1 instruction more expensive on the match case, which can very slightly pessimize things if we always have an early match. However, we can't really predict whether a match will happen early or not and so its better to handle the general case that SIMD is good for, of processing longer inputs (and therefore not expecting an early match), as the overall cost is then much lower by doing it this way.